### ISSUE TITLE
Incorrect normals with skinned model since v1.16.0

### ISSUE THREAD

**Issue by wes-nz** [google/filament#5185]
Since #4939 some normals appear to be incorrect for skinned models.

**< v1.15.2:**
- No bone matrices set, normals are correct 🟢
- Bone matrices set, normals are correct 🟢

**> v1.16.0:**
- No bone matrices set, normals are gone 🔴
- Bone matrices set, normals partially incorrect 🟠

**To Reproduce**
1. Download the sample project for Android [here](https://www.dropbox.com/s/sqbqv7uwpa0brpy/Filament%20-%20NormalMap.zip?dl=1)
2. In `build.gradle` change the version of filament to one of the following `1.15.0`, `1.15.2`, `1.16.0`, `1.18.0` - material compilation automatic for those versions.
3. `MainActivity.kt` `l.102` comment / uncomment the matrices operation.

**Expected behavior**
The normals should be similar to version 1.15.0. 
GLTF Viewer show the normals as 1.15.0.

**Screenshots**
![normal](https://user-images.githubusercontent.com/2945380/153514040-a475cf2d-f203-4312-b128-fbd11df3d245.jpg)

**Desktop (please complete the following information):**
 - OS: Android
 - GPU: Adreno 530
 - Backend: OpenGL

**Smartphone (please complete the following information):**
 - Device: Pixel 1
 - OS: Android 10

**Additional context**

The material used for the normal 

```Swift
material {
    name : normal,
    shadingModel : unlit,
    depthWrite: true,
    requires: [
        uv0,tangents
    ]
}

fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);
        material.baseColor.rgb = getWorldNormalVector();
    }
}
```

**Comment by wes-nz** [comment]
Thanks @pixelflinger, I will give it a try whenever I can. 

Do you have any thoughts about the normal at the bottom end of the model?
![neck](https://user-images.githubusercontent.com/2945380/153555842-fd1b71cf-708a-4576-9ec2-bc528671bf4a.jpg)

**Comment by wes-nz** [comment]
@pixelflinger unfortunately the bug is still there in v1.19.0 on Android. The issue should be re-open.

I have updated [the sample project](https://www.dropbox.com/s/z59cgs92rczdejd/Filament%20-%20NormalMap%20-%202.zip?dl=1) to include the latest release of filament.

I will create a new issue regarding my previous message.

**Comment by romainguy** [comment]
@prideout and @bejado as well

**Comment by romainguy** [comment]
Hmm looks like this change is *not* in 1.19.0, it will be in 1.20.0. We cut a release once a week but we leave it in a branch for 1 week, with this fix having been checked in 7 days ago, it would not be in 1.19.0.

@wes-nz Please try with tip of tree or wait for 1.20.0.

**Comment by wes-nz** [comment]
Ok my bad, reading the release note I assumed it was. I will try the tip of the tree soon.

**Comment by wes-nz** [comment]
No luck unfortunately, I built `main` which includes [d21f092](https://github.com/google/filament/commit/d21f092645b8e1e312307cbf89f1484891347c63).

I've updated the sample project with a debug build from `main`, [available here](https://www.dropbox.com/s/fiuca8pdssjggdb/Filament%20-%20NormalMap%20-%203.zip?dl=0). Thanks.

**Comment by romainguy** [comment]
@pixelflinger can you take a look when you have a moment please?
